### PR TITLE
[BREAKING] Switch default nodes to eks managed node groups.

### PR DESCRIPTION
**Pull request by niroz89**
- Change core nodes to managed
- Cleanup outputs
- Disable node termination handler by default

### CODE REVIEW

File: outputs.tf [hunk 1]
```diff
@@ -57,8 +52,3 @@ output "ebs_kms_key_arn" {
   description = "KMS Key ARN used for EBS encryption"
   value       = module.kms_ebs.key_arn
 }
-
-output "node_termination_handler_sqs_arn" {
-  description = "ARN of the SQS queue used to handle node termination events"
-  value       = module.node_termination_handler_sqs.sqs_queue_arn
-}
```
Comment (lawliet89) [inline outputs.tf:64]: Shouldn't remove this. Output null if it's not enabled.

Comment (niroz89) [reply outputs.tf:64]: creation of `node_termination_handler_sqs_arn` for default core nodes is removed that's why this output is removed. 

`essentials` submodule also will not create nth by default, if we are provisioning self_managed node groups using submodule, then we have to create sqs separately and pass the arn to the essentials submodule.

Comment (niroz89) [reply outputs.tf:64]: should I create the sqs as part of `essentials` ?

it's created here, right now.

https://github.com/SPHTech-Platform/terraform-aws-eks/blob/8b355c29008eaae7baf636c86e3a0f0d93c4611b/node_groups.tf#L113

Comment (lawliet89) [reply outputs.tf:64]: Sure.

**Comment by thepoppingone** [comment]
@niroz89 could we set the default managed node group to use the bottlerocket ami?

File: node_groups.tf [hunk 2]
```diff
@@ -6,17 +6,16 @@ locals {
     launch_template_use_name_prefix = true
     launch_template_name            = var.default_group_launch_template_name
 
-    platform      = "bottlerocket"
-    ami_id        = coalesce(var.default_group_ami_id, data.aws_ami.eks_default_bottlerocket.id)
-    instance_type = var.default_group_instance_type
+    platform       = "bottlerocket"
+    ami_id         = coalesce(var.default_group_ami_id, data.aws_ami.eks_default_bottlerocket.id)
```
Comment (niroz89) [inline node_groups.tf:10]: @thepoppingone here we are already using bottlerocket for default group.

Comment (thepoppingone) [reply node_groups.tf:10]: somehow, if we use the submodule (eks_managed_nodes) without the cluster module, the submodule is still using Amazon Linux 2. I just tried to create a new node group (without a new cluster) the default remains to be not bottlerocket

Comment (lawliet89) [reply node_groups.tf:10]: @niroz89 he means the default in the submodule.

**Review by lawliet89 — APPROVED** [review]

File: node_groups.tf (hunk 2 above)
Comment (niroz89) [reply node_groups.tf:10]: @thepoppingone default bottlerocket is set at the submodule too.

**Pull request merged**